poollet/machinepoollet/mcm/generic.go

```diff
@@ -41,7 +42,7 @@ type Generic struct {
 	listener sets.Set[*listener]
 
 	machineClassByName         map[string]*iri.MachineClassStatus
-	machineClassByCapabilities map[capabilities][]*iri.MachineClassStatus
+	machineClassByCapabilities map[*iri.MachineClassStatus]capabilities
```

Comment:
some thoughts on this change:

Nitpick: this map isn't tracking the machineclasses by capabilities anymore because key and value got switched. Name should reflect this to avoid confusion, something like `machineClassCapabilities`.

1. when we change the key to the `MachineClassStatus` we are adding considerable "waste" from the grpc message structs. Does it make sense to only use the field we consider as important, `MachineClassStatus.MachineClass`?